python-lib/osisoft_plugin_common.py

```diff
@@ -244,8 +245,19 @@ def remove_unwanted_columns(row):
 
 def format_output(input_row, reference_row=None, is_enumeration_value=False):
     output_row = copy.deepcopy(input_row)
+    type_column = None
     if reference_row:
+        if type_column:
+            reference_row["Type"] = type_column
         output_row.update(reference_row)
+    if "Value" in output_row and isinstance(output_row.get("Value"), dict):
+        type_column = output_row.get("Type")
+        output_row = output_row.get("Value")
```

Comment:
Bug: This line overwrites the reference row data - I think the output_row.update(reference_row) needs to be moved till later.

On a related note - it would be nice to retain both the Type of the attribute and the type of the summary in different columns (need to be careful about naming and retaining consistent behaviour). I will try some different approaches to solve both things.

Reply:
Suggestion for this is to move the `if reference_row:` section to the end of the method and make it like this, and sure there is only one return statement. I think this will prevent the bug with the attribute search connector, and for Summary data requests it will retain the existing `Type` column (which is the attribute type, i.e. a data type like string or double or whatever) and add another column called `SummaryType`.

```
    if reference_row:
        if type_column:
            reference_row["SummaryType"] = type_column
        output_row.update(reference_row)
```

Reply:
To make this work nicely you would also need to add `SummaryType` into  `SCHEMA_ATTRIBUTES_METRICS_RESPONSE` in the osisoft_constants.py

Reply:
👍 Fixed in [fa72118](https://github.com/dataiku/dss-plugin-osisoft/pull/40/commits/fa72118eee213aa13b5ed9936331b985630bc3e6). I'm not adding `SummaryType` for now since that changes the schema and would block existing flow / scenarios